include/swift/Parse/SyntaxParsingContext.h

```diff
@@ -112,6 +113,9 @@ class alignas(1 << SyntaxAlignInBits) SyntaxParsingContext {
   // Reference to the
   SyntaxParsingContext *&CtxtHolder;
 
+  /// A cache of nodes that can be reused when creating the current syntax tree
+  SyntaxParsingCache *SyntaxParsingCache = nullptr;
```

Comment:
We could move this instance to `RootContextData`. It will make it clear that all `SyntaxParsingContext` share a single cache entry. Same with `Arena`.